tasks/pools.yml

```diff
@@ -0,0 +1,25 @@
+---
+
+- name: Setup rsyslog pool configuration
+  template:
+    src: 'etc/rsyslog.d/pool.conf.j2'
+    dest: '/etc/rsyslog.d/{{ item.name }}.conf'
+    owner: '{{ item.owner | default("root") }}'
+    group: '{{ item.group | default("root") }}'
+    mode: '{{ item.mode | default("0644") }}'
+    backup: '{{ item.backup | default("yes")  }}'
+  with_items: [ '{{ rsyslog_pools }}' ]
+  notify:
+    - Reload rsyslogd
```

Comment:
Actually, it's best to hook an element of a list on something that is required for this to work, in this case `item.name` is needed, otherwise Ansible won't know where to put the snippet. So:
```
when: ((item.name is defined and item.name) and
       (item.enabled is undefined or not item.enabled))
```